src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,152 @@
+    private Path dataDir = Paths.get(currentDir, "src", "main", "data");
+    private Path fileDir = Paths.get(currentDir, "src", "main", "data", "list.txt");
+    /**
+     * Checks if the /data directory exists.
+     *
+     * @return True if it exists, False otherwise.
+     */
+    private boolean dirExists() {
+        return Files.exists(dataDir);
+    }
+
+    /**
+     * Checks if the /data/list.txt file exists.
+     *
+     * @return True if it exists, False otherwise.
+     */
+    private boolean fileExists() {
+        return Files.exists(fileDir);
+    }
+
+    /**
+     * Private method to create the list.txt file in the /data directory.
+     *
+     * @return list.txt File
+     */
+    private File createFile() {
+        File newFile = new File(dataDir.toString(), "list.txt");
+        System.out.println("Creating File...");
+        try {
+            newFile.createNewFile();
+            System.out.println("Save file created at: " + this.fileDir);
+            return newFile;
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    /**
+     * Private method that gets the list.txt file if present, else a new
+     * file is created.
+     *
+     * @return list.txt File
+     */
+    private File getFile() {
+        if (fileExists()) {
+            return fileDir.toFile();
+        } else if (dirExists()) {
+            return createFile();
+        } else {
+            dataDir.toFile().mkdirs();
+            return createFile();
+        }
+    }
+
+    /**
+     * Updates the list.txt File with the current list. If list.txt file is not
+     * present, a new file is created.
+     *
+     * @param list The TaskList that is used to update the list.
+     * @return True if update is successful, False otherwise.
+     */
+    public boolean updateFile(TaskList list) {
+        try {
+            System.out.println("Saving changes...");
+            File file = getFile();
+            FileWriter writer = new FileWriter(file);
+            for (Task t : list.getList()) {
+                writer.write(t.getSaveString());
+                writer.write("\n");
+            }
+            writer.close();
+            System.out.println("Changes saved.");
+            return true;
+        } catch (IOException e) {
+            e.printStackTrace();
+            System.out.println("Something went wrong during saving!");
+            return false;
+        }
+    }
+
+    /**
+     * Gets the TaskList that has been saved in list.txt File.
+     *
+     * @return TaskList.
+     */
+    public TaskList getList() {
+        try {
+            TaskList list = new TaskList();
+            Scanner scanner = new Scanner(this.getFile());
+            while (scanner.hasNextLine()) {
+                Matcher matcher = pattern.matcher(scanner.nextLine());
+                if (matcher.find()) {
+                    boolean done = matcher.group(2).equals("1");
```

Comment:
The naming would be better to set as isDone instead of done, as it is a boolean data type. It is just so that the variable is more descriptive